src/main/java/com/softawii/capivara/listeners/TwitterListener.java

```diff
@@ -0,0 +1,53 @@
+package com.softawii.capivara.listeners;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Component;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Component
+public class TwitterListener extends ListenerAdapter {
+    private final String patternStrTwitter;
+    private final String patternStrX;
+    public TwitterListener(JDA jda) {
+        this.patternStrTwitter = "https://twitter.com/(\\w+)/status/(\\d+)";
+        this.patternStrX = "https://x.com/(\\w+)/status/(\\d+)";
+        jda.addEventListener(this);
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.getAuthor().isBot()) return;
+
+        String message = event.getMessage().getContentRaw();
+
+        if (message.startsWith("https://twitter.com/")){
```

Comment:
Não vale a pena checar direto se da match com o pattern? Se for o caso da pra retornar um Optional

Reply:
Uma alternativa é pensar num estilo TryGetValue (https://docs.oracle.com/en/database/oracle/oracle-database/23/odpnt/ConnectionStringBuilderTryGetValue.html#GUID-F9AE5F27-244B-4D5B-828A-321FF7B9E1FA)

Retorno = Sucesso
Variável Out = URL tratada